### PR TITLE
fix pwfs tutorial

### PR DESCRIPTION
**Pull request by syhaffert**
I changed the pyramid wavefront sensor tutorial to make it compatible with the new pwfs interface. It now runs again.

### CODE REVIEW

File: doc/tutorial_notebooks/PyramidWFS/PyramidWFS.ipynb [hunk 1]
```diff
@@ -304,51 +299,53 @@
    "source": [
     "def create_closed_loop_animation():\n",
     "    \n",
-    "    wf_ref = Wavefront(magellan_aperture, wavelength_wfs)\n",
-    "    PSF = propagator.forward(wf_ref).power\n",
-    "    PSF /= PSF.max()\n",
+    "    PSF = prop( deformable_mirror(wf) ).power\n",
```
Comment (ehpor) [inline doc/tutorial_notebooks/PyramidWFS/PyramidWFS.ipynb:302]: Fix spaces after parentheses.

File: doc/tutorial_notebooks/PyramidWFS/PyramidWFS.ipynb [hunk 2]
```diff
@@ -304,51 +299,53 @@
...
-    "    im2 = imshow_field(image_ref, pwfs.output_grid)\n",
+    "    im2 = imshow_field(image_ref, pwfs_grid)\n",
+    "    plt.colorbar()\n",
     "    \n",
     "    plt.subplot(1,3,3)\n",
     "    plt.title('Science image plane')\n",
-    "    im3 = imshow_field(np.log10(PSF), vmin=-5, vmax=0)\n",
-    "    \n",
-    "    plt.close(fig)\n",
+    "    im3 = imshow_field( np.log10(PSF / norm), vmax=0, vmin=-5, cmap='inferno')\n",
```
Comment (ehpor) [inline doc/tutorial_notebooks/PyramidWFS/PyramidWFS.ipynb:317]: Fix spaces after parentheses.

File: doc/tutorial_notebooks/PyramidWFS/PyramidWFS.ipynb [hunk 3]
```diff
@@ -304,51 +299,53 @@
...
     "        deformable_mirror.actuators = (1-leakage) * deformable_mirror.actuators - gain * reconstruction_matrix.dot(diff_image)\n",
     "\n",
     "        phase = magellan_aperture * deformable_mirror.surface\n",
     "        phase -= np.mean(phase[magellan_aperture>0])\n",
     "        \n",
-    "        psf = propagator.forward(wf_dm).power\n",
-    "        im1.set_data(*pupil_grid.separated_coords, (magellan_aperture * deformable_mirror.surface).shaped)\n",
-    "        im2.set_data(*pwfs.output_grid.separated_coords, wfs_image.shaped)\n",
-    "        im3.set_data(*focal_grid.separated_coords, np.log10(psf.shaped/psf.max()))\n",
+    "        psf = prop( deformable_mirror(wf) ).power\n",
```
Comment (ehpor) [inline doc/tutorial_notebooks/PyramidWFS/PyramidWFS.ipynb:336]: Fix spaces after parentheses.

File: doc/tutorial_notebooks/PyramidWFS/PyramidWFS.ipynb [hunk 4]
```diff
@@ -304,51 +299,53 @@
...
+    "        \n",
+    "        im1.set_data(*pupil_grid.separated_coords, (magellan_aperture * deformable_mirror.surface).shaped / 1e-6)\n",
+    "        im2.set_data(*pwfs_grid.separated_coords, wfs_image.shaped)\n",
+    "        im3.set_data(*focal_grid.separated_coords, np.log10( psf.shaped / norm ) )\n",
```
Comment (ehpor) [inline doc/tutorial_notebooks/PyramidWFS/PyramidWFS.ipynb:340]: Fix spaces after parentheses.

**Review by ehpor — CHANGES_REQUESTED** [review]
Just the same PEP8 formatting error four times in a row. Otherwise looks perfect to me.

For a future PR: I saw that this tutorial still uses the old way of animation. That should be switched over sometime to the new way (see the EFC tutorial for what I mean with "the new way"). No priority on that at all; just wanted to put it down somewhere.

**Comment by syhaffert** [comment]
Yeah, I went through the tutorial and there are many things I would do differently now. I think at a certain point we should probably overhaul a couple of these tutorials and make them uniform?

**Review by ehpor — APPROVED** [review]

**Pull request merged**